packages/prisma-client/prisma/migrations/20230610111903_button_children/migration.ts

```diff
@@ -0,0 +1,131 @@
+import { PrismaClient } from "./client";
+
+type Text = {
+  type: "text";
+  value: string;
+};
+
+type Id = {
+  type: "id";
+  value: string;
+};
+
+export type Instance = {
+  type: "instance";
+  id: string;
+  component: string;
+  label?: string;
+  children: Array<Id | Text>;
+};
+
+type InstancesList = Instance[];
+
+type BaseProp = {
+  id: string;
+  instanceId: string;
+  name: string;
+  required?: boolean;
+};
+
+type Prop = BaseProp &
+  (
+    | { type: "number"; value: number }
+    | { type: "string"; value: string }
+    | { type: "boolean"; value: boolean }
+    | { type: "asset"; value: string }
+    | {
+        type: "page";
+        value: string | { pageId: string; instanceId: string };
+      }
+    | { type: "string[]"; value: string[] }
+  );
+
+type PropsList = Prop[];
+
+export default async () => {
+  const client = new PrismaClient({
+    // Uncomment to see the queries in console as the migration runs
+    // log: ["query", "info", "warn", "error"],
+  });
+
+  await client.$transaction(
+    async (prisma) => {
+      const chunkSize = 1000;
+      let cursor: undefined | string = undefined;
+      let hasNext = true;
+      while (hasNext) {
+        const builds = await prisma.build.findMany({
+          take: chunkSize,
+          orderBy: {
+            id: "asc",
+          },
+          ...(cursor
+            ? {
+                skip: 1, // Skip the cursor
+                cursor: { id: cursor },
+              }
+            : null),
+        });
+        cursor = builds.at(-1)?.id;
+        hasNext = builds.length === chunkSize;
+
+        for (const build of builds) {
+          const buildId = build.id;
+          try {
+            const instancesList: InstancesList = JSON.parse(build.instances);
+            const propsList: PropsList = JSON.parse(build.props);
+            const newPropsList: PropsList = [];
+            const buttonInstanceIds = new Set<string>();
+            const buttonInnerTexts = new Map<string, string>();
+
+            for (const instance of instancesList) {
+              if (
+                instance.component === "Button" ||
+                instance.component === "VimeoPlayButton"
```

Comment:
btw that brings me to the question: how would we migrate someone else's components and my guess is we can't, if they upgrade component they have to redo them or the component author has to create a migration?
wdyt?

Reply:
I guess similar to what we planned with sdk. Library migrations.

Reply:
true, forgot about it